tools/samtools.wdl

```diff
@@ -332,6 +387,101 @@ task subsample {
     }
 }
 
+task filter {
+    # TODO expose more `view` options
+    meta {
+        description: "Filters a BAM based on its bitwise flag value."
+        help: "This task is a wrapper around `samtools view`. This task will fail if there are no reads in the output BAM. This can happen either because the input BAM was empty or because the supplied `bitwise_filter` was too strict. This task is not suitable for random subsampling. If you want to subsample a BAM, use the `subsample` task instead."
+    }
+
+    parameter_meta {
+        bam: "Input BAM format file to filter"
+        bitwise_filter: "A set of 4 possible read filters to apply. This is a `FlagFilter` object (see ../data_structures/flag_filter.wdl for more information)."
```

Comment:
I don't want to have a detailed explanation of this struct every task it's used, but this I imagine this documentation leads to a bad user experience (particularly as this task doesn't have a default, so _all_ users _must_ go and read another file).

I don't have a solution, I guess I'm just complaining. Any thoughts on this issue?

Reply:
I actually don't mind this. I also do not want to copy the explanation to every usage. Filtering is a fairly manual process anyway and it seems reasonable to require users to read the docs first.

Reply:
That's a good point. I haven't made a ton of headway on `sprocket doc` yet, but I've been toying with the idea of internal links. That would be really cool to add, and this is a prime use case. So I guess I'll leave this as is for now. And I'll revisit if/when I add the capability to link between Markdown files.